### PR TITLE
ref(crons): Produce back-up clock pulses across all partitions

### PR DESCRIPTION
**Pull request by evanpurkhiser**
Part of https://github.com/getsentry/sentry/issues/55821

Since each kafka partition will now have it's own clock we need to ensure a backup clock-pulse (heart beat) is generated across all partitions to keep the clock from stalling in any one partition.

This updates the task responsible for this to do this across all partitions.

### CODE REVIEW

**Review by wedamija — COMMENTED** [review]
This is reasonably sensitive, I think we should have tests that at least mock out the producer and verify things

**Comment by evanpurkhiser** [comment]
Yeah will do, I think the test for this is failing. Just wanted to get it put out there

**Review by wedamija — COMMENTED** [review]
lgtm otherwise

File: src/sentry/monitors/tasks.py [hunk 1]
```diff
@@ -46,7 +52,21 @@ def _get_monitor_checkin_producer() -> KafkaProducer:
...
+    admin_client = AdminClient(conf)
+    result = admin_client.list_topics(topic=topic)
+    topic_metadata = result.topics.get(topic)
+
+    assert topic_metadata
+    return topic_metadata.partitions
+
+
 _checkin_producer = SingletonProducer(_get_monitor_checkin_producer)
+_checkin_partitions = _get_monitor_checkin_partitions()
```
Comment (wedamija) [inline src/sentry/monitors/tasks.py:69]: Hmm are you sure you want this to run on import of this module? Shouldn't it happen on the first time we need to use it then be cached?

Comment (evanpurkhiser) [reply src/sentry/monitors/tasks.py:69]: Yeah good question. Probably don't want this happening all the time.

Is there a simple util for cache on first call? I think there's a `@cached` decorator or something right?

Comment (evanpurkhiser) [reply src/sentry/monitors/tasks.py]: Using `lru_cache`

File: src/sentry/monitors/tasks.py [hunk 2]
```diff
@@ -38,15 +45,29 @@
 MONITOR_TASKS_LAST_TRIGGERED_KEY = "sentry.monitors.last_tasks_ts"
 
 
-def _get_monitor_checkin_producer() -> KafkaProducer:
+def _get_producer() -> KafkaProducer:
```
Comment (evanpurkhiser) [inline src/sentry/monitors/tasks.py:48]: just renamed it since the other one has such a long name

**Review by wedamija — APPROVED** [review]

**Review by lynnagara — APPROVED** [review]

**Comment by evanpurkhiser** [comment]
Added a test for `_get_partitions` but had to remove it for now since I don't think we always run kafka in test

**Pull request merged**